tests/test_0437-stream-of-many-json-files.py

```diff
@@ -0,0 +1,143 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/master/LICENSE
+
+from __future__ import absolute_import
+
+import sys
+import os
+import json
+
+import pytest
+import numpy
+
+import awkward1
+
+def test_tostring():
+    # write a json string from an array built from
+    # multiple json fragments from a string
+    str = """{"x": 1.1, "y": []}
+             {"x": 2.2, "y": [1]}
+             {"x": 3.3, "y": [1, 2]}
+             {"x": 4.4, "y": [1, 2, 3]}
+             {"x": 5.5, "y": [1, 2, 3, 4]}
+             {"x": 6.6, "y": [1, 2, 3, 4, 5]}"""
+
+    array = awkward1.from_json(str)
+    assert awkward1.to_list(array) == [
+        [{'x': 1.1, 'y': []}],
+        [{'x': 2.2, 'y': [1]}],
+        [{'x': 3.3, 'y': [1, 2]}],
+        [{'x': 4.4, 'y': [1, 2, 3]}],
+        [{'x': 5.5, 'y': [1, 2, 3, 4]}],
+        [{'x': 6.6, 'y': [1, 2, 3, 4, 5]}]]
+
+    assert awkward1.to_json(array) == '[[{"x":1.1,"y":[]}],[{"x":2.2,"y":[1]}],[{"x":3.3,"y":[1,2]}],[{"x":4.4,"y":[1,2,3]}],[{"x":5.5,"y":[1,2,3,4]}],[{"x":6.6,"y":[1,2,3,4,5]}]]'
+
+def test_fromstring():
+    # read multiple json fragments from a string
+    str = """{"x": 1.1, "y": []}
+             {"x": 2.2, "y": [1]}
+             {"x": 3.3, "y": [1, 2]}
+             {"x": 4.4, "y": [1, 2, 3]}
+             {"x": 5.5, "y": [1, 2, 3, 4]}
+             {"x": 6.6, "y": [1, 2, 3, 4, 5]}"""
+
+    array = awkward1.from_json(str)
+    assert awkward1.to_list(array) == [
+        [{'x': 1.1, 'y': []}],
+        [{'x': 2.2, 'y': [1]}],
+        [{'x': 3.3, 'y': [1, 2]}],
+        [{'x': 4.4, 'y': [1, 2, 3]}],
+        [{'x': 5.5, 'y': [1, 2, 3, 4]}],
+        [{'x': 6.6, 'y': [1, 2, 3, 4, 5]}]]
+
+def test_array_tojson():
+    # convert float 'nan' and 'inf' to user-defined strings
+    array = awkward1.layout.NumpyArray(numpy.array([
+        [float('nan'), float('nan'), 1.1],
+        [float('inf'), 3.3, float('-inf')]]))
+
+    assert awkward1.to_json(array, nan_string='NaN', infinity_string='inf',
+        minus_infinity_string='-inf') == '[["NaN","NaN",1.1],["inf",3.3,"-inf"]]'
+
+    array2 = awkward1.Array([[0, 2], None, None, None, None, None])
+    assert awkward1.to_json(array2, nan_string='NaN') == '[[0,2],"NaN","NaN","NaN","NaN","NaN"]'
```

Comment:
Tests like this one will have to change when the rule is applied above that `nan_string` replaces floating-point NaN, but not Python None.

```python
>>> float("nan"), None
(nan, None)
```

Reply:
done